### PR TITLE
[DOC] Relateds for combo methods

### PR DESCRIPTION
**Pull request by BurdetteLamar**

### CODE REVIEW

File: array.c [hunk 1]
```diff
@@ -4938,7 +4938,7 @@ rb_ary_fill(int argc, VALUE *argv, VALUE ary)
  *    a = [0, 1] + [2, 3]
  *    a # => [0, 1, 2, 3]
  *
- *  Related: Array#concat, Array#*.
+ *  Related: see {Methods or Combining}[rdoc-ref:Array@Methods+for+Combining].
```
Comment (peterzhu2118) [inline array.c:4941]:
```suggestion
 *  Related: see {Methods for Combining}[rdoc-ref:Array@Methods+for+Combining].
```

File: array.c [hunk 2]
```diff
@@ -5457,7 +5457,7 @@ ary_make_hash_by(VALUE ary)
  *  Element are compared using method <tt>#eql?</tt>
  *  (as defined in each element of +self+).
  *
- *  Related: Array#difference.
+ *  Related: see {Methods or Combining}[rdoc-ref:Array@Methods+for+Combining].
```
Comment (peterzhu2118) [inline array.c:5460]: Ditto

File: array.c [hunk 3]
```diff
@@ -5563,12 +5563,7 @@ rb_ary_difference_multi(int argc, VALUE *argv, VALUE ary)
  *  Identifies common elements using method <tt>#eql?</tt>
  *  (as defined in each element of +self+).
  *
- *  Related:
- *
- *  - Array#intersection: intersection of +self+ and multiple other arrays.
- *  - Array#|: union of +self+ and one other array.
- *  - Array#union: union of +self+ and multiple other arrays.
- *
+ *  Related: see {Methods or Combining}[rdoc-ref:Array@Methods+for+Combining].
```
Comment (peterzhu2118) [inline array.c:5566]: Ditto

File: array.c (hunk 1 above)
Comment (BurdetteLamar) [reply array.c]: Fixed.

File: array.c (hunk 2 above)
Comment (BurdetteLamar) [reply array.c]: Fixed.

File: array.c (hunk 3 above)
Comment (BurdetteLamar) [reply array.c]: Fixed.

**Review by peterzhu2118 — APPROVED** [review]

**Pull request merged**